web/backend/src/Server.ts

```diff
@@ -216,10 +216,33 @@ app.post('/api/add_role', (req, res) => {
   const { sciper } = req.body;
   const { role } = req.body;
 
-  usersDB.put(sciper, role).catch((error) => {
-    res.status(500).send('Failed to add role');
-    console.log(error);
-  });
+  // The sciper has to contain 6 numbers
+  if (sciper > 999999 || sciper < 100000) {
+    res.status(400).send('Sciper length is incorrect');
+    return;
+  }
+
+  // Call https://search-api.epfl.ch/api/ldap?q=228271, if the answer is
+  // empty then sciper unknown, otherwise add it in userDB
+  axios
+    .get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`)
+    .then((response) => {
+      if (response.data.length === 0) {
+        res.status(400).send('Unknown Sciper');
+        return;
+      }
+
+      usersDB.put(sciper, role).catch((error) => {
+        res.status(500).send('Failed to add role');
+        console.log(error);
+      });
+
+      res.status(200).send('Role added');
```

Comment:
this will also execute in the case of the `catch` above, don't you want to put in a `then` (before the `catch`) ?